backend/src/main/java/com/bakdata/conquery/models/auth/conquerytoken/ConqueryTokenRealm.java

```diff
@@ -0,0 +1,128 @@
+import com.bakdata.conquery.models.identifiable.ids.specific.UserId;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.dropwizard.util.Duration;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.authc.ExpiredCredentialsException;
+import org.apache.shiro.authc.IncorrectCredentialsException;
+
+@Slf4j
+public class ConqueryTokenRealm extends ConqueryAuthenticationRealm {
+
+	private static final Class<? extends AuthenticationToken> TOKEN_CLASS = JwtToken.class;
+
+	private final MasterMetaStorage storage;
+	
+	@Setter
+	private JWTConfig jwtConfig = new JWTConfig();
+	
+	
+	public ConqueryTokenRealm(MasterMetaStorage storage) {
+		this.storage = storage;
+		setAuthenticationTokenClass(TOKEN_CLASS);
+		setCredentialsMatcher(new SkippingCredentialsMatcher());
+	}
+
+	@Override
+	protected ConqueryAuthenticationInfo doGetConqueryAuthenticationInfo(AuthenticationToken token) throws AuthenticationException {
+		if (!(TOKEN_CLASS.isAssignableFrom(token.getClass()))) {
+			log.trace("Incompatible token. Expected {}, got {}", TOKEN_CLASS, token.getClass());
+			return null;
+		}
+		log.trace("Token has expected format.", TOKEN_CLASS, token.getClass());
+		DecodedJWT decodedToken = null;
+		try {
+			decodedToken = jwtConfig.getTokenVerifier(this).verify((String) token.getCredentials());
+		}
+		catch (TokenExpiredException e) {
+			log.trace("The provided token is expired.");
+			throw new ExpiredCredentialsException(e);
+		}
+		catch (SignatureVerificationException | InvalidClaimException e) {
+			log.trace("The provided token was not successfully verified against its signature or claims.");
+			throw new IncorrectCredentialsException(e);
+		}
+		catch (JWTVerificationException e) {
+			log.trace("The provided token could not be verified.");
+			throw new AuthenticationException(e);
+		}
+		catch (Exception e) {
+			log.trace("Unable to decode token", e);
+			throw new AuthenticationException(e);
+		}
+		log.trace("Received valid token.");
+
+		String username = decodedToken.getSubject();
+
+		UserId userId = UserId.Parser.INSTANCE.parse(username);
+		User user = storage.getUser(userId);
+		// try to construct a new User if none could be found in the storage
+		if (user == null) {
+			log.warn(
+				"Provided credentials were valid, but a corresponding user was not found in the System. You need to add a user to the system with the id: {}",
+				userId);
+			return null;
+		}
+
+		return new ConqueryAuthenticationInfo(userId, token, this);
+	}
+	
+
+	
+	public String createTokenForUser(UserId userId) {
+		if(storage.getUser(userId) == null) {
+			throw new IllegalArgumentException("Cannot create a JWT for unknown user with id: " + userId);
+		}
+		return TokenHandler.createToken(userId.toString(), jwtConfig.getJwtDuration(), getName(), jwtConfig.getTokenSignAlgorithm());
```

Comment:
die Duration würde ich hier in eine Java Duration umwandeln, das ist nur ne coole klasse für parsing